src/java.base/share/classes/java/util/Locale.java

```diff
@@ -230,40 +230,26 @@
  * implementations in a Java Runtime Environment might not support any
  * particular Unicode locale attributes or key/type pairs.
  *
- * <h3>Creating a Locale</h3>
+ * <h3><a id="ObtainingLocale">Obtaining a Locale</a></h3>
  *
- * <p>There are several different ways to create a {@code Locale}
+ * <p>There are several ways to obtain a {@code Locale}
  * object.
  *
  * <h4>Builder</h4>
  *
  * <p>Using {@link Builder} you can construct a {@code Locale} object
  * that conforms to BCP 47 syntax.
  *
- * <h4>Constructors</h4>
+ * <h4>Factory Method</h4>
  *
- * <p>The {@code Locale} class provides three constructors:
- * <blockquote>
- * <pre>
- *     {@link #Locale(String language)}
- *     {@link #Locale(String language, String country)}
- *     {@link #Locale(String language, String country, String variant)}
- * </pre>
- * </blockquote>
- * These constructors allow you to create a {@code Locale} object
- * with language, country and variant, but you cannot specify
- * script or extensions.
- *
- * <h4>Factory Methods</h4>
- *
- * <p>The method {@link #forLanguageTag} creates a {@code Locale}
+ * <p>The method {@link #forLanguageTag} obtains a {@code Locale}
```

Comment:
The factory name `forLanguageTag` is a bit off-putting, it doesn't seem like the best name for the factory.
Yes, it already exists and does what's required but you might get better uptake with a more natural name.

Some alternatives:
 - `Locale.of("en_US")` - short and conventional
 - `Locale.ofLanguage("en_US")` - 'of' prefix is used in other factories
 - `Locale.forLanguage("en_US")` - natural but less conventional

Reply:
I was thinking of a *new* factory method, along the line with Stuart's suggestion, something like this:
```
Locale.of(String... elements)
```
where elements can either `(lang)`, `(lang, ctry)`, `(lang, ctry, vrnt)`, or `(lang, ctry, vrnt, scpt)`. Either element can be an empty string, but cannot be null. These elements are *not* language tags, but conventional arguments to constructors, so it is compatible (and works as a stop-gap) to the old constructors. This way third parties will not have to deal with the boilerplate code mentioned above on migration.